src/MarkedOptions.ts

```diff
@@ -0,0 +1,218 @@
+import type { Token, Tokens, TokensList } from './Tokens.ts';
+import { _Parser } from './Parser.ts';
+import { _Lexer } from './Lexer.ts';
+import { _Renderer } from './Renderer.ts';
+import { _Tokenizer } from './Tokenizer.ts';
+
+export interface SluggerOptions {
+  /** Generates the next unique slug without updating the internal accumulator. */
+  dryrun?: boolean;
+}
+
+export type Rule = RegExp | string;
+
+export interface Rules {
+  [ruleName: string]: Pick<RegExp, 'exec'> | Rule | Rules;
+}
```

Comment:
Because e.g. `block.normal`, `block.gfm` etc. is a Rules object nested within the block rules object. Therefore the recursion.

I moved the `Rule` and `Rules` types to `rules.ts`